### ISSUE TITLE
[JENKINS-35795] API(s) for acknowledging request for user input in a pipeline

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#2763]
<p>A pipeline can pause and ask for user input at any point. </p>

<p>This needs to be exposed via the blue ocean REST api - both that it is waiting, but also that it can take a POST of data to acknowledge it. The input may take the form of Yes/No or it may be a richer set of input. </p>

<p>In Scope: </p>
<ul>
	<li>API to fetch any pipelines blocked on "input" (for current user)</li>
	<li>API to fetch what input is required from user (so a form can be built up on the client)</li>
	<li>API to post confirmation from user (binary yes/no)</li>
	<li>API to take a post of data from user (for non yes/no)<br/>
(these are not necessarily separate unrelated apis, just that there needs to be support for those types of features)</li>
</ul>

<p>Out of scope: <br/>
   (perhaps we should start with simple yes/no)</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35795" target="_blank">API(s) for acknowledging request for user input in a pipeline</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vivek.png" title="vivek's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek">vivek</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Critical
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: api, blueocean-imported
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-12-26T07:41:13+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>A pipeline can pause and ask for user input at any point. </p>

<p>This needs to be exposed via the blue ocean REST api - both that it is waiting, but also that it can take a POST of data to acknowledge it. The input may take the form of Yes/No or it may be a richer set of input. </p>

<p>In Scope: </p>
<ul>
	<li>API to fetch any pipelines blocked on "input" (for current user)</li>
	<li>API to fetch what input is required from user (so a form can be built up on the client)</li>
	<li>API to post confirmation from user (binary yes/no)</li>
	<li>API to take a post of data from user (for non yes/no)
(these are not necessarily separate unrelated apis, just that there needs to be support for those types of features)</li>
</ul>

<p>Out of scope: 
   (perhaps we should start with simple yes/no)</p>

</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35795?focusedId=261378&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261378">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>See <a href="https://cloudbees.atlassian.net/browse/UX-390" class="external-link" target="_blank" rel="nofollow noopener">https://cloudbees.atlassian.net/browse/UX-390</a> 
I am not sure if there is a spec for the type of input that pipeline can support. I use the snippet builder to see what options there are (try it). </p>

<p>The current stage view in Jenkins 2.0 only supports yes/no, but I think we need to support a bit more (not sure how this affects the API though, maybe not at all as it can just pass though). </p>
</pre>
</details>

</ul>
</details>

<p>See <a href="https://cloudbees.atlassian.net/browse/UX-390" class="external-link" target="_blank" rel="nofollow noopener">https://cloudbees.atlassian.net/browse/UX-390</a> <br/>
I am not sure if there is a spec for the type of input that pipeline can support. I use the snippet builder to see what options there are (try it). </p>

<p>The current stage view in Jenkins 2.0 only supports yes/no, but I think we need to support a bit more (not sure how this affects the API though, maybe not at all as it can just pass though). </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35795?focusedId=261304&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261304">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> I think this, and the linked front end ticket, may require some more investigation as to what exactly is required. The current stage viewer can only say "yes/no" to input, pipeline allows more types of input. Do we support them all? A common user request is in the "approval" popover, that pipeline authors could put in some URL/html info (which may depend on context of execution) providing the approver with more context (it's still a yes/no answer, but more info). Thoughts? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> I think this, and the linked front end ticket, may require some more investigation as to what exactly is required. The current stage viewer can only say "yes/no" to input, pipeline allows more types of input. Do we support them all? A common user request is in the "approval" popover, that pipeline authors could put in some URL/html info (which may depend on context of execution) providing the approver with more context (it's still a yes/no answer, but more info). Thoughts? </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-38494%22">JENKINS-38494</a>]</i>
